app/facade/station_facade.rb

```diff
@@ -0,0 +1,11 @@
+class StationFacade
+  def self.create_stations(location)
+    stations = StationService.get_stations(location)
+    stations.map do |station|
+      # require 'pry'; binding.pry 
+      Station.new(station)
+      #why am i getting undefined method 'map' for nilclass error
```

Comment:
This facade is set up properly from what I can tell -- the issue here for the feature spec test that was failing was around the lack of `location` passed through if you immediate visit `/search` in the test. If you go directly to that path without filling out the form where you select an address at the root path, you will get to this point in the code without having a location parameter (more notes in a comment on that test)

Reply:
The other failures for the facade tests and other tests were just VCR errors, because some of the tests had :vcr and some did not. If you want to test things out with a real HTTP connection first, you can always add this line to your config in rails_helper, which will allow you to make a real network call while you're building out the tests. This will allow you to TDD with real network calls and then once things look okay, you can add VCR to mock them so that you're not longer hitting the API
`config.allow_http_connections_when_no_cassette = true`

You've got your tests started solidly. You mentioned feeling like you're jumping around a lot and this is a common thing students run into. Your routine is ultimately your choice based on  what makes sense to you, but this is what I personally do. 
- Start with a very skeletal feature spec to get routes/controllers/paths sorted out. 
- Once the foundation is there, you can either build everything out in the controller or start building out the Facade, service, PORO right away. The following breakdown is if you do this second option: building the components out right away instead of everything in the controller
- Examine the network request to identify domain object I will care about - in this case, Stations. Build out the PORO with TDD using one of the Station items from Postman as the test data. Pull out the attributes I will need in the PORO
- Build out the service using TDD and *using a real network call* (since we're not refactoring but building this out to start, I find this helpful so that I can see when my service call fails due to wrong parameters, missing key, etc). Run the service test *on its own* as I'm doing this. 
- Then build out the facade with TDD focusing out the ultimate output that I want. In this case, it's one closest station
- Return to feature spec to build out the expectations of what I want to see on the page. Build out the view template according to the test/user story. 

In all of these cases, I find it helpful to run one test file at a time in order to not get overwhelmed. Once things are looking good on the class I'm working on, I might run all of them to make sure I didn't break anything else. Generally, when encountering a confusing but, I suggest running only that test in order to isolate the issue and not get distracted by other failures.